File: PAP2223/python/test_mymath.py
import unittest
import mymath

class TestMyMath(unittest.TestCase):
    
    def test_estPair(self):
        self.assertEqual(True,mymath.estPair(20))
        self.assertEqual(False,mymath.estPair(21))
        self.assertEqual(True,mymath.estPair(0))
        self.assertRaises(AssertionError,mymath.estPair,-1)
        self.assertRaises(AssertionError,mymath.estPair,5.0)
        self.assertRaises(AssertionError,mymath.estPair,"brol") 

if __name__ == "__main__":
    unittest.main()

File: PAP2223/test/tp01/TestMyMath.java
package tp01;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;
import static org.junit.jupiter.api.Assertions.assertThrows;

import org.junit.jupiter.api.Test;

class TestMyMath {
	@Test
	void testEstPair() {
		assertTrue(MyMath.estPair(8));
		assertFalse(MyMath.estPair(3));
		assertTrue(MyMath.estPair(0));
		assertThrows(AssertionError.class, () -> MyMath.estPair(-2));
	}
}
